rpc/handler.go

```diff
@@ -112,11 +124,21 @@ func (h *handler) handleBatch(msgs []*jsonrpcMessage) {
 		return
 	}
 	// Process calls on a goroutine because they may block indefinitely:
+	responseBytes := 0
 	h.startCallProc(func(cp *callProc) {
 		answers := make([]*jsonrpcMessage, 0, len(msgs))
-		for _, msg := range calls {
+		for idx, msg := range calls {
 			if answer := h.handleCallMsg(cp, msg); answer != nil {
 				answers = append(answers, answer)
+				// Once total size of responses exceeds an allowed maximum,
+				// generate error messages for all remaining calls and stop further processing
+				if responseBytes += len(answer.Result); responseBytes > BatchResponseMaxSize {
```

Comment:
Technically `+=` is not thread safe.
I would recommend to use https://pkg.go.dev/sync/atomic#Int32.Add

Reply:
Nevermind, it's sequential on a thread not multi-thread

Reply:
`append` isn't thread-safe either. Is this an upstream bug or am I missing something that avoids problems here?